### DISCUSSION TITLE
Provide tail -f on collection

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#15208] [Q&A]
We have lots of large collections which function as queues. When I open the collection in the raven studio it provides a nice list ordered by time with the last entered at the top of the list.

Can this be reproduced in c#.  I've tried doing a query with a take(1000) but the sorting is not in creation order.

**Comment by krivulcik** [comment]
Sounds like Data Subscriptions is exactly what you need, doesn't it?

https://ravendb.net/docs/article-page/5.4/Csharp/client-api/data-subscriptions/what-are-data-subscriptions

**Reply by garayx** [reply]
There is a streaming feature as well https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/loading-entities#example-i-2 it should return you the entities ordered by `@last-modified`

**Reply by bobskigit** [reply]
Not sure if I articulated the question well. I just want the first n items from a large collection order by @last-modified. Same behaviour as in the studio. 

You click a collection and it opens a grid to display the items, it refreshes every n seconds.

If I do a streaming query

var enumerator = await session
    .Advanced
    .StreamAsync(session.Query<AdRequest>().Take(1000));

It does not give me the first 1000 items in @last-modified order

**Reply by mcrio** [reply]
Have you tried something along these lines:

```
.OrderByDescending(doc => doc["@metadata"]["@last-modified"])
```
